conan/tools/meson/toolchain.py

```diff
@@ -447,11 +447,11 @@ def _filter_list_empty_fields(v):
     def _sanitize_env_format(value):
         if value is None or isinstance(value, list):
             return value
+        assert isinstance(value, str)
```

Comment:
I think that we could raise a ConanException instead.

Reply:
Done